lxd/instances_get.go

```diff
@@ -430,23 +437,26 @@ func doInstancesGet(d *Daemon, r *http.Request) (interface{}, error) {
 			}
 		} else {
 			threads := 4
-			if len(instanceNames) < threads {
-				threads = len(instanceNames)
+			if len(instancesAndProjects) < threads {
+				threads = len(instancesAndProjects)
 			}
 
-			queue := make(chan string, threads)
+			queue := make(chan [2]string, threads)
 
 			for i := 0; i < threads; i++ {
 				wg.Add(1)
 
 				go func() {
 					for {
-						instanceName, more := <-queue
+						instanceAndProjectNames, more := <-queue
 						if !more {
 							break
 						}
 
-						inst, found := nodeInstances[instanceName]
+						instanceName := instanceAndProjectNames[0]
+						projectName := instanceAndProjectNames[1]
+
+						inst, found := nodeInstances[uniqueInstance(instanceName, projectName)]
```

Comment:
If we update nodeInstances to be keyed on the same value as is returned from `queue` this would be easier and safer.